lib/node_runner.go

```diff
@@ -548,15 +549,21 @@ func (nr *NodeRunner) proposeNewBallot(roundNumber uint64) error {
 		VotingHole:     VotingNOTYET,
 	}
 
-	{
-		err := sebakcommon.RunChecker(transactionsChecker, sebakcommon.DefaultDeferFunc)
-		if err != nil {
-			if _, ok := err.(sebakcommon.CheckerErrorStop); !ok {
-			}
+	err := sebakcommon.RunChecker(transactionsChecker, sebakcommon.DefaultDeferFunc)
+	if err != nil {
+		return err
+	}
+
+	// remove invalid transactions
+	var invalidTransactions []string
+	for _, hash := range availableTransactions {
+		if _, found := sebakcommon.InStringArray(transactionsChecker.ValidTransactions, hash); found {
```

Comment:
This part is crucial, right? Because this function is called every block height. So we should focus on the performance.
I figure it would be better to use transactionsChecker.ValidTransactionsMap instead of transactionsChecker.ValidTransactions because of #214 :)